pyls/plugins/rope_rename.py

```diff
@@ -31,13 +30,19 @@ def pyls_rename(config, workspace, document, position, new_name):
                 'uri': uris.uri_with(
                     document.uri, path=os.path.join(workspace.root_path, change.resource.path)
                 ),
+                'version': document.version 
```

Comment:
This is incorrect, the `uri` we're returning is derived from the rope resource (line above), however this version number is derived from the document within which the rename was triggered.

If you check the LSP, version is not a required field: https://microsoft.github.io/language-server-protocol/specification#versionedtextdocumentidentifier

However, we can do better than just never setting it. If you use `workspace.get_document(doc_uri)` and then use that document's version then it will return None (correctly) if the server has never received a didOpen message, or the correct version number if it has. 

My concern with doing this is that rope doesn't operate on the in-memory view of the documents. We should probably fix that before making too heavy usage of rope. In particular, the edits returned by rope could be inconsistent with the version of the document.

Reply:
So is it incorrect that the client is requiring the version? I think they're only accidentally requiring it, but I don't know for sure. 

Regarding rope: yes, it would be nice if we could get it to use the in-memory buffer. In my earlier work with rope I just had the client save the file, but that was the lazy way out.

Reply:
Yup, I think it's incorrect of the client to require it. Though it might be correct if they only require it for documents that the client has sent a didOpen message for